### PR TITLE
Unwrap ValueTask<T> return types

### PR DESCRIPTION
**Pull request by alasdaircs**
Unwrap ValueTask<T> return types, same as Task<T>
Unwrap (Value)Task<ActionResult<T>>
Unify unwrapping for consistency
Tests included
See #4373

### CODE REVIEW

File: src/NSwag.Generation/GenericResultWrapperTypes.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		private readonly static String[] WrapperTypeNames =
+		{
+			"Task`1",
+			"ValueTask`1",
+			"JsonResult`1",
+			"ActionResult`1"
+		};
+
+		internal static bool IsGenericWrapperType(string typeName)
+			=> WrapperTypeNames.Contains( typeName );
```
Comment (lahma) [inline src/NSwag.Generation/GenericResultWrapperTypes.cs:18]: this is quite slow, probably better to just do 
``` 
=> typeName == "Task`1" || typeName == "ValueTask`1" || typeName== "JsonResult`1" || typeName == "ActionResult`1" 
```

Comment (alasdaircs) [reply src/NSwag.Generation/GenericResultWrapperTypes.cs:18]: I have no idea what the difference is in practice - I tend to write simple stuff like this more for readability than marginal gains in execution time but I guess it all adds up. I did some searching and found this [Performance insight for Enumerable.Contains](https://learn.microsoft.com/en-us/visualstudio/profiling/perf-insights-contains) which points out the doing this with a HashSet<T> would make it O(1), however for this short list of only fours items I bet your code smaller and possibly faster, so I'm happy to make your suggested change.

**Comment by alasdaircs** [comment]
I believe the failing checks are failing away from my changes - I think master I based on at 3ee148af had the same failing tests.

**Comment by lahma** [comment]
Yes I think we need to wait for some PRs to land into master before many of the builds succeed...

**Review by lahma — APPROVED** [review]

**Pull request merged**